tactics/hints.ml

```diff
@@ -66,6 +66,9 @@ let decompose_app_bound t =
     | Proj (p, c) -> ConstRef (Projection.constant p), Array.cons c args
     | _ -> raise Bound
 
+let secvars_of_hyps hyps =
+  List.fold_left (fun l id -> Id.Pred.add id l) Id.Pred.empty hyps
+
```

Comment:
You are right, it should do that. And in general we can make the patch less expensive by checking if we actually are in a section or not.